### DISCUSSION TITLE
Is it able to get the full collection from the IChangeSet<T>?

### DISCUSSION THREAD

**Discussion by Snailya** [reactivemarbles/DynamicData#889] [Q&A] [answered]
I want to add a side effect to my tree node. Whenever a node's item has a property of IsACopyFrom, i need to add the children of that id to this item too. I think this could be done using CustomAdaptor that implements IChangeSetAdaptor<T>. But I don't know how to get the full collection from IChangeSet<T>, so I have no ways to find the children by node id.

**Comment by JakenVeina** [comment, marked as answer]
There's no magic trick to retrieving the full collection from an `IChangeSet<>` you just have to rebuild your own copy of the collection, as each changeset comes through. That's what pretty-much all the DD operators do. Depending on the type of changeset we're talking about, there should be a `.Clone()` extension method you can use to apply a changeset to a collection.
